### DISCUSSION TITLE
`isDirty` exported from `useForm`

### DISCUSSION THREAD

**Discussion by AdrianMrn** [inertiajs/inertia#521] [Ideas]
@reinink would you be open to having an `isDirty`/`isEdited`/… variable exported from the `useForm` hook, that is `true` when a value has changed since a successful submit?

This is useful for adding "navigate away listeners":

```
const navigateAwayRef = useRef();
useEffect(() => {
    if (isDirty) {
        navigateAwayRef.current = Inertia.on('before', () =>
            confirm('You have unsaved changes, are you sure you want to navigate away?')
        );
    }

    if (!isDirty && navigateAwayRef.current) {
        navigateAwayRef.current();
    }
}, [isDirty]);
```

If you like the idea, I could submit a PR that adds this.

**Comment by reinink** [comment]
Yeah, I was kind of thinking we'd probably eventually want to add this to the form helper. Feel free to PR. 👍
